client/src/Features/MicrosoftLearn/MicrosoftLearnPage.tsx

```diff
@@ -18,16 +18,20 @@ import {
 } from './MicrosoftLearnSelectedItemsList';
 import { getCommonSpacingStyle } from './MicrosoftLearnStyles';
 import { pagesDisplayNames } from '../../Router/Consts';
+import { useLocation } from 'react-router-dom';
 
 type MicrosoftLearnPageStyles = SimpleComponentStyles<'root' | 'separator'>;
 
 const MicrosoftLearnPageInner = ({ styles }: IStylesOnly<MicrosoftLearnPageStyles>): JSX.Element => {
   const learnStore = useStore('microsoftLearnStore');
+  const location = useLocation();  
+  const qsParams = new URLSearchParams(location.search);
+
   useEffect(() => {
     if (learnStore.catalog == null) {
-      learnStore.initializeCatalog();
+      learnStore.initializeCatalog(qsParams);
     }
-  }, [learnStore]);
+  }, [learnStore, qsParams]);

```

Comment:
Why have we passed this to `useEffect`? I don't believe this would ever be changing, hence, we would never be re-initializing the `learnStore` catalog. It's fine if there's some error in build without this.

Reply:
Yeah the compiler was unhappy with removing any parameter that was being used inside the hook from the dependency list; and would have also caused the build to fail!